zh/enable-admission-webhook.md

```diff
@@ -0,0 +1,103 @@
+---
+title: 开启 TiDB-Operator 准入控制器
+summary: 介绍如何开启 TiDB-Operator 准入控制器以及它的作用
+category: how-to
+---
+
+# TiDB Operator 准入控制器
+
+Kubernetes 在 1.9 版本引入了 [动态准入机制](https://kubernetes.io/zh/docs/reference/access-authn-authz/extensible-admission-controllers/)， 从而使得拥有对 Kubernetes 中的各类资源进行修改与验证的功能。 在 TiDB Operator 中，我们也同样使用了动态准入机制来帮助我们进行相关资源的修改、验证与运维。
+
+## 开启 TiDB Operator 准入控制器
+
+TiDB Operator 在默认安装情况下不会开启准入控制器，你需要手动开启:
+
+1. 修改 Operator 的 `values.yaml`
+
+    开启 Operator Webhook 特性:
+
+    ```yaml
+    admissionWebhook:
+      create: true
+    ```
+
+2. 配置失败策略
+
+    在 Kubernetes 1.15 版本之前，动态准入机制的管理机制的粒度较粗并且并不方便去使用。所以为了防止 TiDB Operator 的动态准入机制影响全局集群，我们需要配置 [失败策略](https://kubernetes.io/zh/docs/reference/access-authn-authz/extensible-admission-controllers/#failure-policy)。
+
+    对于 Kubernetes 1.15 以下的版本，我们推荐将 TiDB Operator 失败策略配置为 `Ignore`，从而防止 TiDB Operator 的 admission webhook 出现异常时影响整个集群。
+
+    ```yaml
+    ......
+    failurePolicy:
+        validation: Ignore
+        mutation: Ignore
+    ```
+
+    对于 Kubernetes 1.15 及以上的版本，我们推荐给 TiDB Operator 失败策略配置为 `Failure`，由于 Kubernetes 1.15 及以上的版本中，动态准入机制已经有了基于 Label 的筛选机制，所以不会出现 TiDB Operator 的 admission webhook 出现异常时影响整个集群。
+
+    ```yaml
+    ......
+    failurePolicy:
+        validation: Failure
+        mutation: Failure
+    ```
+
+3. 安装/更新 TiDB Operator
+
+    修改完 `values.yaml` 文件中的上述配置项以后，进行 TiDB Operator 部署或者更新。安装与更新 TiDB Operator 请参考[在 Kubernetes 上部署 TiDB Operator](deploy-tidb-operator.md)。 
+
+
+## TiDB Operator 准入控制器功能
+
+TiDB Operator 通过准入控制器的帮助实现了许多功能。我们将在这里介绍各个资源的准入控制器与其相对应的功能。
+
+1. Pod 验证准入控制器:
+
+    Pod 准入控制器提供了对 PD/TiKV/TiDB 组件安全下线与安全上线的保证，通过 Pod 验证准入控制器，我们可以通过它实现 [重启 Kubernetes 上的 TiDB 集群](restart-a-tidb-cluster.md)，该组件在准入控制器开启的情况下默认开启。
+
+    ```yaml
+    admissionWebhook:
+    validation:
+        pods: true
+    ```
+
+2. StatefulSet 验证准入控制器:
+
+    StatefulSet 验证准入控制器帮助实现 TiDB 集群中 TiDB/TiKV/PD 组件的灰度发布，该组件在准入控制器开启的情况下默认关闭。
```

Comment:
Give more detail here? How to use the canary release?

Reply:
updated.